packages/ld-key-pair/KeyPair.ts

```diff
@@ -0,0 +1,103 @@
+/**
+ * defines the ability to sign based on a KeyPair or only
+ * from a private key
+ */
+interface Signer {
+    privateKey: string;
+    keyPair?: KeyPair;
+    sign(options: {
+        data: any;
+    }): string;
+}
+
+/**
+ * defines the ability to perform verification based on a KeyPair
+ * or a public key
+ */
+interface Verifier {
+    publicKey: string;
+    keyPair?: KeyPair;
+    verify(options: {
+        data: any;
+        signature: string;
+    }): boolean;
+}
+
+/**
+ * public key side of a key pair
+ */
+interface PublicNode {
+    id: string;
+    type: any;
+    controller?: any;
+    publicKey?: string;
+    publicKeyBase58?: string;
+}
+/**
+ * public key side of a key pair
+ */
+ interface PrivateNode {
+    id: string;
+    type: any;
+    controller?: any;
+    privateKey?: string;
+    privateKeyBase58?: string;
+}
+
+/**
+ * defines the common properties and methoods in use by 
+ * key pairs in the LD realm based on existing implementations
+ */
+interface KeyPair {
+    id: string;
+    type: any;
+    controller?: any;
+    publicKey?: string;
+    privateKey?: string;
+    publicKeyBase58?: string;
+    privateKeyBase58?: string;
```

Comment:
IMO these "encoded version" like base58, multibase or jwk should only be on externals only... for example,

k = KeyPair.from(k1: ExternalKeyPair) => k.publicKey, where k1.publicKeyBase58 and NOT k1.publicKey

I think we can make publicKey required.

Reply:
agreed - i think pub can and should be required, but went back and forth a few times on that